### PR TITLE
Add support for other normalizations (i.e., GroupNorm) in ResNet

### PR DESCRIPTION
**Pull request by fmassa**
There is currently no code-path leveraging GroupNorm (or any other normalization).

Using GroupNorm (which has one extra parameter) could be achieved as follows:

```python
def get_gn(num_channels):
    return nn.GroupNorm(32, num_channels)

model = models.ResNet(models.Bottleneck, [3, 4, 6, 3], norm_layer=get_gn)
```

### CODE REVIEW

**Comment by ekagra-ranjan** [comment]
Should these changes be made to other torchvision models as well?

**Comment by soumith** [comment]
@ekagra-ranjan yes they should, but I believe francisco is on a sprint to do them already

**Comment by rvarun7777** [comment]
Are there any pre-trained weights for resnet with GroupNorm?

**Comment by fmassa** [comment]
@rvarun7777 not yet. Models with GroupNorm are particularly useful when training a model from scratch, and do not necessarily bring better accuracies compared to BatchNorm (for classification)
